src/gestures.c

```diff
@@ -326,18 +327,22 @@ static void touch_detect_update(
 	int i = 0, 
 	touching = 0;
 	struct timeval expire;
-	foreach_bit(i, ms->touch_used) {
-		microtime(&expire);
-		timeraddms(&ms->touch[i].down, cfg->button_expire, &expire);
-		if ((cfg->button_move || cfg->button_expire == 0 || timercmp(&ms->touch[latest].down, &expire, <)) &&
-			!(cfg->ignore_thumb && GETBIT(ms->touch[i].flags, MT_THUMB)) &&
-			!(cfg->ignore_palm && GETBIT(ms->touch[i].flags, MT_PALM)) &&
-			!(GETBIT(ms->touch[i].flags, MT_EDGE))) {
-			touching++;
-			LOG_EMULATED("touch_detect_udpate: latest >=0: touching++ =%d\n", touching);
+
+	/*If latest is not set, the finger placement is not valid*/
+	if (latest >= 0) {
+		foreach_bit(i, ms->touch_used) {
```

Comment:
This if is a tiny optimization to bypass the `for loop` if `latest` is not set. If it is not set, there is no need to count the number of finger since there is virtually no finger on the pad.